vms/platformvm/blocks/builder/camino_network.go

```diff
@@ -53,20 +56,59 @@ func (n *caminoNetwork) CrossChainAppRequest(_ context.Context, chainID ids.ID,
 		return errUnknownCrossChainMessage // this would be fatal
 	}
 
-	tx, err := n.txBuilder.NewRewardsImportTx()
+	if err := n.appSender.SendCrossChainAppResponse(
+		ctx,
+		chainID,
+		requestID,
+		[]byte(n.caminoRewardMessage()),
+	); err != nil {
+		n.ctx.Log.Error("caminoCrossChainAppRequest failed to send response", zap.Error(err))
+		// we don't want fatal here: response is for logging only, so
+		// its better to not respond properly, than crash the whole node
+		return nil
+	}
+
+	return nil
+}
+
+func (n *caminoNetwork) caminoRewardMessage() string {
+	tx, err := n.newRewardsImportTx()
 	if err != nil {
-		n.ctx.Log.Error("caminoCrossChainAppRequest couldn't create rewardsImportTx", zap.Error(err))
-		return nil // we don't want fatal here
+		return err.Error()
+	}
+
+	utx, ok := tx.Unsigned.(*txs.RewardsImportTx)
+	if !ok {
+		// should never happen
+		err = fmt.Errorf("unexpected tx type: expected *txs.RewardsImportTx, got %T", utx)
+		n.ctx.Log.Error("caminoCrossChainAppRequest failed to create rewardsImportTx", zap.Error(err))
+		return fmt.Sprintf("caminoCrossChainAppRequest failed to issue rewardsImportTx: %s", err)
 	}
 
 	n.ctx.Lock.Lock()
 	defer n.ctx.Lock.Unlock()
 
 	if err := n.blkBuilder.AddUnverifiedTx(tx); err != nil {
-		n.ctx.Log.Error("caminoCrossChainAppRequest couldn't add rewardsImportTx to mempool", zap.Error(err))
-		// we don't want fatal here: its better to have network running
-		// and try to repair stalled reward imports, than crash the whole network
+		n.ctx.Log.Error("caminoCrossChainAppRequest failed to add unverified rewardsImportTx to block builder", zap.Error(err))
+		return fmt.Sprintf("caminoCrossChainAppRequest failed to add unverified rewardsImportTx to block builder: %s", err)
 	}
 
-	return nil
+	amounts := make([]uint64, len(utx.Ins))
+	for i := range utx.Ins {
+		amounts[i] = utx.Ins[i].In.Amount()
+	}
+
+	return fmt.Sprintf("caminoCrossChainAppRequest issued rewardsImportTx with utxos with %v nCAM", amounts)
+}
+
+func (n *caminoNetwork) newRewardsImportTx() (*txs.Tx, error) {
+	n.ctx.Lock.Lock()
+	defer n.ctx.Lock.Unlock()
+
+	tx, err := n.txBuilder.NewRewardsImportTx()
+	if err != nil {
+		n.ctx.Log.Error("caminoCrossChainAppRequest failed to create rewardsImportTx", zap.Error(err))
+		return nil, fmt.Errorf("caminoCrossChainAppRequest failed to create rewardsImportTx: %w", err)
```

Comment:
Just as a side note: I was a bit surprised to see %w here and looked it up. Seems that Errorf has a special handling with %w for Error types. So this is completely fine. 
Raises the question whether we should change that also to other instances where Errors are returned via fmt (like above in line 93).

But that's then something we can do in Berlin-1+ - if you think it makes sense please create an issue for that.